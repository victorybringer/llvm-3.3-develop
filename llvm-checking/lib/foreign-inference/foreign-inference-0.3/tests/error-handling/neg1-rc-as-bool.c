#include <sys/ioctl.h>
#include <errno.h>

int ioctlWrapper(int fd, int ionum) {
  char buffer[100];
  int r = ioctl(fd, ionum, buffer);
  if(r) {
    if(errno == EINVAL)
      return -100;
    else if(errno == EBUSY)
      return -20;
    else if(errno == ENODEV)
      return -5;

    return -99;
  }

  return 0;
}
