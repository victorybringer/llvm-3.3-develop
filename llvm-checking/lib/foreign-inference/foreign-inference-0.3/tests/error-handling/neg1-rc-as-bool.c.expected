fromList [ ("ioctlWrapper", (fromList [ ], ReturnConstantInt (fromList [-5, -20, -99, -100])))
         ]
